src/native/thrift/nativeConnection.ts

```diff
@@ -9,22 +9,32 @@ import {
     ITransportConstructor,
     IProtocolConstructor,
     TApplicationException,
-    TApplicationExceptionType
+    TApplicationExceptionType,
+    BufferedTransport,
+    CompactProtocol,
+    ThriftProcessor
 } from '@creditkarma/thrift-server-core'
 
 import * as uuid from 'uuid';
+import * as Webview from 'mobile-apps-thrift-typescript/Webview';
+import { WebviewHandler } from 'native/webviewApi';
 
 declare global {
     interface Window {
         nativeConnections: { [id: string]: NativeConnection };
-        AndroidWebViewMessage?: (nativeMessage: NativeMessage) => {};
+        AndroidWebViewRequest?: (nativeMessage: NativeMessage) => {};
+        AndroidWebViewResponse?: (nativeMessage: NativeMessage) => {};
         webkit: {
             messageHandlers: {
-                iOSWebViewMessage: {
+                iOSWebViewRequest: {
+                    postMessage: (nativeMessage: NativeMessage) => {};
+                };
+                iOSWebViewResponse: {
                     postMessage: (nativeMessage: NativeMessage) => {};
                 };
             };
         };
+        receiveNativeMessage: (nativeMessage: NativeMessage) => void;
```

Comment:
Should `NativeMessage` be renamed to `Message` as it's used both ways now. Maybe the `connectionId` should be optional or we have two separate types?

Reply:
I think the connection id should always be required. Probably makes sense to rename to `Message` or maybe `ThriftMessage`?